libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -231,6 +232,51 @@ bool StaticAnalyzer::visit(InlineAssembly const& _inlineAssembly)
 	return true;
 }
 
+bool StaticAnalyzer::visit(BinaryOperation const& _operation)
+{
+	if (
+		_operation.rightExpression().annotation().isConstant &&
+		(_operation.getOperator() == Token::Div || _operation.getOperator() == Token::Mod)
+	)
+		if (auto rhs = dynamic_pointer_cast<RationalNumberType const>(
```

Comment:
These are one of the first uses of `dynamic_pointer_cast`. Are we moving over to it from `dynamic_cast`?

Reply:
I count 10 occurrences in current ``develop`` already...
And not all occurrences of ``dynamic_cast`` can be replaced by ``dynamic_pointer_cast`` or vice-versa. At times you only have a raw pointer and have to use ``dynamic_cast`` and at times you have to extend the life-time of a ``shared_ptr``, so you have to use ``dynamic_pointer_cast``....
So I think it's justified to have both in the codebase.

Reply:
Here for example the ``shared_ptr`` returned by ``evaluate`` will only be shared with a member of ``ConstantEvaluator``, so if both fall out of scope, the object will be destroyed, so I have to use ``dynamic_pointer_cast`` to extend its lifetime.